manifests/squid.pp

```diff
@@ -0,0 +1,24 @@
+# The class to manage squid. This is used by pulp streamer.
+class pulp::squid(
+  Integer $port = 3128,
+  String $maximum_object_size = '5 GB',
+  String $maximum_object_size_in_memory = '100 MB',
+) {
+  class { '::squid3':
+    use_deprecated_opts           => false,
+    http_port                     => [ "${port} accel defaultsite=127.0.0.1:8751" ],
+    acl                           => [ "Safe_ports port ${port}" ],
+    http_access                   => [ 'allow localhost', 'deny to_localhost', 'deny all' ],
+    cache                         => [ 'allow all' ],
+    maximum_object_size           => $maximum_object_size,
+    maximum_object_size_in_memory => $maximum_object_size_in_memory,
+    cache_dir                     => [ 'aufs /var/spool/squid 10000 16 256' ],
```

Comment:
Fancy making this configurable while you are at it? Not required.

Reply:
That's kind of the point of this PR. Even if it's only via hiera. The next step would be to switch over to https://github.com/voxpupuli/puppet-squid which is supported. Do you happen to know some good acceptance tests to verify it works?